pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java

```diff
@@ -269,6 +280,38 @@ public void startController(Map<String, Object> properties)
     configAccessor.set(scope, Helix.DEFAULT_HYPERLOGLOG_LOG2M_KEY, Integer.toString(12));
   }
 
+  protected void waitForController()
```

Comment:
there's a `TestUtils.waitForCondition` that does most of the time keeping similar to this one. can we reuse that?

Reply:
I don't think they are the same. `waitForCondition(Function<Void, Boolean> condition, ...)` family aborts if an exception is thrown. Also, this method logs in a different way if it was an exception. Info logs are not printed by default. I think it may be useful in future to have the logs if the flaky error show up again.